subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,486 @@
+# -*- coding: utf-8 -*-
+import json
+import logging
+import os
+import re
+
+from babelfish import Language, language_converters
+from datetime import datetime
+from guessit import guessit
+from rarfile import RarFile, is_rarfile
+from requests import Session
+from tempfile import NamedTemporaryFile
+from zipfile import ZipFile, is_zipfile
+
+from . import ParserBeautifulSoup, Provider
+from ..cache import region, SHOW_EXPIRATION_TIME
+from ..exceptions import AuthenticationError, ConfigurationError
+from ..subtitle import Subtitle, fix_line_ending, guess_matches, sanitized_string_equal
+from ..video import Episode, Movie, SUBTITLE_EXTENSIONS
+
+TIMEOUT = 10
+
+logger = logging.getLogger(__name__)
+
+
+class LegendasTvSubtitle(Subtitle):
+    provider_name = 'legendastv'
+
+    def __init__(self, language, page_link, subtitle_id, name, binary_content, imdb_id=None, type=None, season=None,
+                 year=None, no_downloads=None, rating=None, featured=False, multiple_episodes=False, timestamp=None):
+        super(LegendasTvSubtitle, self).__init__(language, page_link=page_link)
+        self.subtitle_id = subtitle_id
+        self.name = name
+        self.imdb_id = imdb_id
+        self.type = type
+        self.season = season
+        self.year = year
+        self.no_downloads = no_downloads
+        self.rating = rating
+        self.featured = featured
+        self.multiple_episodes = multiple_episodes
+        self.timestamp = timestamp
+        self.binary_content = binary_content
+
+    @property
+    def id(self):
+        return '%s-%s' % (self.subtitle_id, self.name.lower())
+
+    def get_matches(self, video, hearing_impaired=False):
+        matches = super(LegendasTvSubtitle, self).get_matches(video, hearing_impaired=hearing_impaired)
+
+        # The best available information about a subtitle is its name. Using guessit to parse it.
+        guess = guessit(self.name, {'type': self.type})
+        # Cleaning release_group: foo[bar] -> foo
+        if guess.get('release_group'):
+            guess['release_group'] = re.sub('\[\w+\]', '', guess['release_group'])
```

Comment:
Yes, I think so. I was expecting this issue to be handled by guessit, but I
got a reply from them saying that this part will not change and we need to
do post-processing:
https://github.com/guessit-io/guessit/issues/196#issuecomment-177627695



On Mon, 1 Feb 2016 at 11:43 Antoine Bertin <notifications@github.com> wrote:

> In subliminal/providers/legendastv.py
> <https://github.com/Diaoul/subliminal/pull/513#discussion_r51402762>:
>
> > +        self.multiple_episodes = multiple_episodes
> > +        self.timestamp = timestamp
> > +        self.binary_content = binary_content
> > +
> > +    @property
> > +    def id(self):
> > +        return '%s-%s' % (self.subtitle_id, self.name.lower())
> > +
> > +    def get_matches(self, video, hearing_impaired=False):
> > +        matches = super(LegendasTvSubtitle, self).get_matches(video, hearing_impaired=hearing_impaired)
> > +
> > +        # The best available information about a subtitle is its name. Using guessit to parse it.
> > +        guess = guessit(self.name, {'type': self.type})
> > +        # Cleaning release_group: foo[bar] -> foo
> > +        if guess.get('release_group'):
> > +            guess['release_group'] = re.sub('\[\w+\]', '', guess['release_group'])
>
> Should that be extended to other providers? Or is it only legendastv
> related?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Diaoul/subliminal/pull/513/files#r51402762>.
>